flink-table-store-common/src/main/java/org/apache/flink/table/store/file/predicate/PredicateBuilder.java

```diff
@@ -88,19 +91,39 @@ public Predicate startsWith(int idx, Object patternLiteral) {
     }
 
     public Predicate leaf(NullFalseLeafBinaryFunction function, int idx, Object literal) {
-        return new LeafPredicate(function, rowType.getTypeAt(idx), idx, singletonList(literal));
+        RowType.RowField field = rowType.getFields().get(idx);
+        return new LeafPredicate(
+                function, field.getType(), idx, field.getName(), singletonList(literal));
     }
 
     public Predicate leaf(LeafUnaryFunction function, int idx) {
-        return new LeafPredicate(function, rowType.getTypeAt(idx), idx, Collections.emptyList());
+        RowType.RowField field = rowType.getFields().get(idx);
+        return new LeafPredicate(
+                function, field.getType(), idx, field.getName(), Collections.emptyList());
     }
 
     public Predicate in(int idx, List<Object> literals) {
-        return new LeafPredicate(In.INSTANCE, rowType.getTypeAt(idx), idx, literals);
+        RowType.RowField field = rowType.getFields().get(idx);
+        LeafPredicate in =
+                new LeafPredicate(In.INSTANCE, field.getType(), idx, field.getName(), literals);
+        if (literals.size() > 20) {
```

Comment:
Add more tests for `IN` predicate. Check if the result is correct with less (more) than 20 literals, and with (or without) nulls.